packages/kit/src/core/build/build_client.js

```diff
@@ -0,0 +1,119 @@
+import fs from 'fs';
+import path from 'path';
+import { svelte } from '@sveltejs/vite-plugin-svelte';
+import { deep_merge } from '../../utils/object.js';
+import { print_config_conflicts } from '../config/index.js';
+import { create_app } from '../create_app/index.js';
+import { copy_assets, posixify } from '../utils.js';
+import { create_build, find_deps } from './utils.js';
+
+/**
+ * @param {{
+ *   cwd: string;
+ *   assets_base: string;
+ *   config: import('types/config').ValidatedConfig
+ *   manifest_data: import('types/internal').ManifestData
+ *   build_dir: string;
+ *   output_dir: string;
+ *   client_entry_file: string;
+ *   service_worker_entry_file: string | null;
+ * }} options
+ */
+export async function build_client({
+	cwd,
+	assets_base,
+	config,
+	manifest_data,
+	build_dir,
+	output_dir,
+	client_entry_file
+}) {
+	create_app({
+		manifest_data,
+		output: build_dir,
+		cwd
+	});
+
+	copy_assets(build_dir);
+
+	process.env.VITE_SVELTEKIT_AMP = config.kit.amp ? 'true' : '';
+
+	const client_out_dir = `${output_dir}/client/${config.kit.appDir}`;
+
+	/** @type {Record<string, string>} */
+	const input = {
+		start: path.resolve(cwd, client_entry_file)
+	};
+
+	// This step is optional — Vite/Rollup will create the necessary chunks
```

Comment:
I wonder if this should be a dev only thing? I've always felt that SvelteKit file names reveal a bit too much about my internal code that I might want to obfuscate a bit more

Reply:
Hmm. There isn't really such a thing as a dev-only build. I don't know if this rises to the level of exposing an option. Personally I definitely err on the side of debuggability over obfuscation (which is no defense against evildoers anyway), so my preference would be to maintain the current behaviour. 

This isn't new to this PR though (it just looks that way because `build_client` was factored out into a new module) — maybe this is worth a separate issue so we don't need to decide it here?

Reply:
i'm fine punting on it since it's not new like you said